detectionmetrics/utils/conversion.py

```diff
@@ -14,7 +14,14 @@ def hex_to_rgb(hex: str) -> Tuple[int, ...]:
     :rtype: Tuple[int, ...]
     """
     hex = hex.strip("#")
-    return tuple(int(hex[i : i + 2], 16) for i in (0, 2, 4))
+    if len(hex) != 6:
+        raise ValueError("Invalid hex code: Must be exactly 6 characters long")
+    
+    try:
+        rgb = tuple(int(hex[i : i + 2], 16) for i in (0, 2, 4))
+        return rgb
+    except:
```

Comment:
this try/except clause is too broad, it would be better to catch ValueError only, which is the type of exception a non-hexadecimal character would raise